### PR TITLE
fix(temp-storage): move workingCache to temporary storage (fixes #1136)

### LINKED ISSUE #1136: Clean up old content
The new Lumi version cleans content and temporary storage on startup to avoid massing old content in the user's data directory. This behavior won't work when there are multiple instances of Lumi.

There are several options: 
- don't allow multi instances (not preferable)
- find a better way of cleaning up the old content (e.g. listen to events, listen to windows close)
- temporarily return to the old behavior (not perform cleanup at all)

### PR DESCRIPTION
**Pull request by JPSchellenberg**
Hey,

I started to work on the issue that the old content is not cleaned up by saving it into the `app.getPath('temp')` folder. Is this a viable way? @sr258

### CODE REVIEW

**Comment by JPSchellenberg** [comment]
I remove the old workingCache directory during the setup process. What do you think?

File: server/src/main.ts [hunk 1]
```diff
@@ -217,7 +218,8 @@ if (!gotSingleInstanceLock) {
         log.info('app is ready');
         const server = await httpServerFactory(
             serverConfigFactory(
-                process.env.USERDATA || app.getPath('userData')
```
Comment (sr258) [inline server/src/main.ts:220]: I think we should reuse the serverConfig from above if possible

File: server/src/main.ts [hunk 2]
```diff
@@ -110,7 +110,8 @@ if (!gotSingleInstanceLock) {
     app.quit();
 } else {
     const serverConfig = serverConfigFactory(
-        process.env.USERDATA || app.getPath('userData')
+        process.env.USERDATA || app.getPath('userData'),
+        process.env.TEMPDATA || app.getPath('temp')
```
Comment (sr258) [inline server/src/main.ts]: The temp path is generic and not app specific. In the player, I've appended a directory:
`const tmpDir = path.join(app.getPath('temp'), 'lumiplayer')`

File: server/src/boot/setup.ts [hunk 3]
```diff
@@ -13,8 +13,17 @@ export default async function setup(
     serverConfig: IServerConfig
 ): Promise<void> {
     try {
+        // If the workingCache (prior 0.8.0) still exists in userData remove it. -> https://github.com/Lumieducation/Lumi/pull/1727
+        const workingCachePath = path.join(
```
Comment (sr258) [inline server/src/boot/setup.ts]: The same should be done for the temporary files

**Review by sr258 — CHANGES_REQUESTED** [review]
The temporary files in the "real" temporary directory aren't deleted at the moment. In LumiPlayer I've done it like this:

```
app.on('will-quit', (event) => {
    log.debug(`Deleting temporary directory: ${tmpDir}`);
    fsExtra.removeSync(tmpDir);
});
```

Relevant commit: https://github.com/Lumieducation/LumiPlayer/commit/e9d0dc69c41da1e3ead7aeeb12bbf30f1378a641#

While we're working on this, it would also make sense to rename the "workingCache". The name is a bit confusing and I would simply call it content storage.

**Review by sr258 — APPROVED** [review]
Everything looks fine. I think it makes sense to include this in 0.8.0 after all.

**Pull request merged**